src/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfo.cs

```diff
@@ -252,11 +247,25 @@ public void CopyRuntimeSettingsTo(JsonPropertyInfo other)
             other._escapedName = _escapedName;
         }
 
+        // Create a property that is either ignored at run-time. It uses typeof(int) in order to prevent any
+        // issues with unsupported types and helps ensure we don't accidently (de)serialize it.
+        public static JsonPropertyInfo CreateIgnoredPropertyPlaceholder(PropertyInfo propertyInfo, JsonSerializerOptions options)
```

Comment:
Why do we need to create such a placeholder at all?

Reply:
Do you mean vs not having metadata at all, or creating a "real" JsonPropertyInfo?

If a property has {JsonIgnore] and don't add the placeholder then it would be added (incorrectly) to the extension data.